common/tools/dev-tool/src/util/admin/migrate-package/codemods/fixSourceFile.ts

```diff
@@ -0,0 +1,97 @@
+  const sourceLinesToRemove = [
+    "const should = chai.should();",
+    "chai.use(chaiAsPromised);",
+    "chai.use(chaiExclude);",
+    "const expect = chai.expect;",
+  ];
+  // Iterate over all the statements in the source file
+  for (const statement of sourceFile.getStatements()) {
+    // Remove old legacy lines
+    for (const line of sourceLinesToRemove) {
+      if (statement.getText() === line) {
+        statement.remove();
+      }
+    }
+
+    const patternsToReplace = [
+      { pattern: /\(this: Context\)/g, replace: "(ctx)" },
+      { pattern: /\(this\.currentTest\)/g, replace: "(ctx)" },
+      { pattern: /\(!this\.currentTest\?\.isPending\(\)\)/g, replace: "(!ctx.task.pending)" },
+      { pattern: /this\.skip\(\);/g, replace: "ctx.skip();" },
+    ];
+
+    // Replace the patterns in the source file
+    for (const { pattern, replace } of patternsToReplace) {
+      if (pattern.test(statement.getText())) {
+        statement.replaceWithText(statement.getText().replace(pattern, replace));
+      }
+    }
+  }
+
+  // Iterate over all the import declarations
+  for (const importExportDeclaration of sourceFile.getImportDeclarations()) {
+    let moduleSpecifier = importExportDeclaration.getModuleSpecifierValue();
+    moduleSpecifier = fixDeclaration(sourceFile, moduleSpecifier);
+    importExportDeclaration.setModuleSpecifier(moduleSpecifier);
+  }
+
+  // iterate over all the export declarations
+  for (const exportDeclaration of sourceFile.getExportDeclarations()) {
+    let moduleSpecifier = exportDeclaration.getModuleSpecifierValue();
+    if (moduleSpecifier) {
+      moduleSpecifier = fixDeclaration(sourceFile, moduleSpecifier);
+      exportDeclaration.setModuleSpecifier(moduleSpecifier);
+    }
+  }
+}
+
+function fixDeclaration(sourceFile: SourceFile, moduleSpecifier: string): string {
+  if (moduleSpecifier.startsWith(".") || moduleSpecifier.startsWith("..")) {
+    if (!moduleSpecifier.endsWith(".js")) {
+      // If the module specifier ends with "/", add "index.js", otherwise add ".js"
+      if (moduleSpecifier.endsWith("/")) {
+        moduleSpecifier += "index.js";
+      } else {
+        // Check if the module specifier is a directory
+        const path = resolve(sourceFile.getDirectoryPath(), moduleSpecifier);
+        if (existsSync(path) && lstatSync(path).isDirectory()) {
+          moduleSpecifier += "/index.js";
+        } else {
+          moduleSpecifier += ".js";
+        }
+      }
+    }
+  }
+  // Fix the node module declaration as well
+  return fixNodeDeclaration(moduleSpecifier);
+}
+
+function fixNodeDeclaration(moduleSpecifier: string): string {
+  const nodeModules = [
+    "assert",
+    "crypto",
+    "events",
+    "fs",
+    "fs/promises",
+    "http",
+    "https",
+    "net",
+    "os",
+    "path",
+    "process",
+    "stream",
+    "tls",
+    "util",
+  ];
+
+  if (nodeModules.includes(moduleSpecifier)) {
+    moduleSpecifier = `node:${moduleSpecifier}`;
```

Comment:
This probably only affects Service Bus but I found that adding "node:" prevents the usage of polyfills when running in browsers.

Reply:
Gotcha! The node: prefix is all existing behavior. We can change it now or separately but I'd hope not to block this PR on it if that's ok?

Reply:
Just a general comment, not blocking at all.